imageio-turbojpeg/src/main/java/de/digitalcollections/turbojpeg/imageio/TurboJpegImageReader.java

```diff
@@ -116,6 +120,23 @@ public Iterator<ImageTypeSpecifier> getImageTypes(int imageIndex) {
         .iterator();
   }
 
+  /**
+   * Calculate the closest value to a given minimum. This function should be used when defining min sizes
+   * of region height or width, because Math.min is not sufficient in rare cases, when the returned minimum
+   * value is smaller than the desired size. Consider this example:
+   * First value is 44 and second value is 112 with a user-specified min value of 100. Math.min would select 44,
+   * which is wrong, because it is under the user-specified threshold of 100.
+   * @param firstValue The first value to be checked
+   * @param secondValue The second value to be checked
+   * @param minValue The minimum value
+   * @return Integer of the closest value w.r.t. a given min value.
+   *         If both values are under the min value, min value will be returned
+   */
+  int getClosestValue(int firstValue, int secondValue, int minValue) {
+    Set<Integer> collectedValues = Stream.of(firstValue, secondValue).filter(val -> val >= minValue).collect(Collectors.toSet());
+    return collectedValues.isEmpty() ? minValue : Collections.min(collectedValues);
```

Comment:
Could this be written as `Math.max(Math.min(firstValue, secondValue), minValue)`?

Reply:
This is e.g. not working with:

```java
int firstValue = 107;
int secondValue = 44;
int minValue = 100;
```

Original function would collect 107 which is right in this case.

Suggested function would return 100 which is not what we want here (100 should only be returned when `firstValue` and `secondValue` return smaller values than `minValue`.

Reply:
I still think that using a Stream for this is probably overkill, this should do the same, but result in much faster execution:

```suggestion
    if (firstValue < minVal && secondValue < minVal) {
      return minVal;
    } else {
      return Math.min(firstVal, secondVal);
    }
```

Reply:
how about this stream-less implementation?

```
if (firstVal < minVal && secondVal < minVal) {
  return minVal;
} else if (firstVal >= minVal) {
  return firstVal;
} else {
  return secondVal;
}
```

Should be much quicker!
```

Reply:
Thanks!